tests/test_directive_other.py

```diff
@@ -148,3 +148,22 @@ def test_toctree_twice(app):
     assert_node(doctree[0][0],
                 entries=[(None, 'foo'), (None, 'foo')],
                 includefiles=['foo', 'foo'])
+
+
+@pytest.mark.sphinx(testroot='directive-include')
+def test_include_source_read_event(app):
+    sources_reported = {}
+
+    def source_read_handler(app, doc, source):
+        sources_reported[doc] = source[0]
+    app.connect("source-read", source_read_handler)
```

Comment:
Btw, is there any issue if you have other `.. include::` directives inside an included document? You could also add a test to reflect the original issue, namely transforming the included content.

Reply:
The test that I added does reflect the original issue: no "source-read" event generated for included RST files.  

I can absolutely add another test to transform the content, and include within an include file.

Reply:
I have now added the test and you were indeed right that includes of includes were problematic with the sub-classing approach.  I have reverted to the patching solution which I used in the original extension I made for Sphinx.